pojo/src/main/java/com/spotify/hamcrest/pojo/IsPojo.java

```diff
@@ -246,28 +247,18 @@ private static SerializedLambda serializeLambda(final Object lambda) {
   }
 
   private static void ensureDirectMethodReference(final SerializedLambda serializedLambda) {
-    final Method targetMethod;
     try {
       final Class<?> implClass = Class.forName(serializedLambda.getImplClass().replace('/', '.'));
-      targetMethod = findMethodByName(implClass, serializedLambda.getImplMethodName());
-    } catch (NoSuchMethodException | ClassNotFoundException e) {
-      throw new IllegalStateException(
-          "serializeLambda returned a SerializedLambda pointing to an invalid class/method", e);
-    }
-
-    if (targetMethod.isSynthetic()) {
-      throw new IllegalArgumentException("The supplied lambda is not a direct method reference");
-    }
-  }
-
-  private static Method findMethodByName(final Class<?> cls, final String methodName)
-      throws NoSuchMethodException {
-    for (final Method method : cls.getDeclaredMethods()) {
-      if (method.getName().equals(methodName)) {
-        return method;
+      if (stream(implClass.getMethods())
+          .noneMatch(m ->
+              m.getName().equals(serializedLambda.getImplMethodName())
+              && !m.isSynthetic())) {
+        throw new IllegalArgumentException("The supplied lambda is not a direct method reference");
```

Comment:
This is not a super-clear error message - I realise it's the same as before, but perhaps it could be improved anyway? I think it's maybe relevant to distinguish the `NoSuchMethodException` from the synthetic method case, but I'm not sure how this 'feels' when using the API. When can you run into one of these two cases, what diagnostics do you get in that case, and how obvious is it for the user what to fix?

Reply:
I'm not sure it is that useful since the `NoSuchMethodException` case that disappeared shouldn't really be possible. Same for the 

> I suppose `SomeClass` should implement `SomeInterface` in the example?

Yes, fixed

Reply:
The only `NoSuchMethodException` case that I can think of is if you pass a method reference to a static method in another class with the signature `static T method(A a)`, e.g:

```java
class Utils {
    static String getValue(SomeClass instance) { return instance.getValue(); }
}

class SomeClass {
    String getValue() { return "something"; }
}

class TestClass {
    @Test
    void testCase() {
        assertThat(
            new SomeClass(),
            is(pojo(SomeClass.class).where(Utils::getValue, equalTo("something"))));
}
```